tests/Unit/ProtocolTestHelpers.hpp

```diff
@@ -0,0 +1,29 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Utilities/ProtocolHelpers.hpp"
+
+namespace detail {
+
+template <typename ConformingType, typename Protocol>
+struct TestProtocolConformanceImpl : std::true_type {
```

Comment:
the value for this struct can just be the `is_conforming_v` from the second assert, which would then be checked below.